dataflow_transfer/utils/transfer.py

```diff
@@ -0,0 +1,75 @@
+import subprocess
+import logging
+import os
+
+logger = logging.getLogger(__name__)
+
+
+def rsync_is_running(src, dst):
+    """Check if rsync is already running for given src and dst."""
+    pattern = f"rsync.*{src}.*{dst}"
+    try:
+        subprocess.check_output(["pgrep", "-f", pattern])
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+def sync_to_hpc(
+    run_path: str,
+    destination: str,
+    rsync_log: str,
+    background: bool,
+    transfer_details: dict = {},
+):
+    """Sync the run to storage using rsync.
+    Skip if rsync is already running on the run."""
+    remote_destination = (
+        transfer_details.get("user")
+        + "@"
+        + transfer_details.get("host")
+        + ":"
+        + destination
+    )
+    command = [
+        "run-one",
+        "rsync",
+        "-au",
+        "--log-file=" + rsync_log,
+        "--chown=" + transfer_details.get("owner"),
+        "--chmod=" + transfer_details.get("permissions"),
+        remote_destination,
+        run_path,
+        destination,
+    ]
+
+    if rsync_is_running(
```

Comment:
What is the reason to both have this pgrep based check on top of run-one? I vaguely recall that its something Alfred added?

Reply:
I also recall that it's because run-one is very specific. Another rsync with slightly different parameters would not be considered the same.

Reply:
Yes, this bit is stolen an modified from the ONT transfer script. It's an extra security measurement, but maybe it's sufficient to just do the pgrep?